From 84ec20af79b2ecc85858d34ad5e5bab1acef3b89 Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Sun, 18 May 2014 18:55:56 +0300
Subject: [PATCH 4/5] use rilproxy as service

* underp b2g permissions
* don't symlink rilproxy to rild
* add rilproxy service

Qualcomm has modified libril to grant gecko direct access to rild.
This would avoid need for rilproxy. To use working prebuilt libril
from stock, we need to bring back rilproxy as a service.

Change-Id: Iad7eb43789a2caeb76de3ea781b3f90fa8653952
---
 include/private/android_filesystem_config.h |    6 +++---
 rootdir/init.rc                             |    9 ++++++---
 2 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/include/private/android_filesystem_config.h b/include/private/android_filesystem_config.h
index 903b485..590af1b 100644
--- a/include/private/android_filesystem_config.h
+++ b/include/private/android_filesystem_config.h
@@ -220,9 +220,9 @@ static const struct fs_path_config android_dirs[] = {
 ** and will allow partial matches.
 */
 static const struct fs_path_config android_files[] = {
-    { 00775, AID_ROOT,      AID_ROOT,      0, "system/b2g/b2g" },
-    { 00775, AID_ROOT,      AID_ROOT,      0, "system/b2g/updater" },
-    { 00775, AID_ROOT,      AID_ROOT,      0, "system/b2g/plugin-container" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/b2g/b2g" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/b2g/updater" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/b2g/plugin-container" },
     { 00440, AID_ROOT,      AID_SHELL,     0, "system/etc/init.goldfish.rc" },
     { 00550, AID_ROOT,      AID_SHELL,     0, "system/etc/init.goldfish.sh" },
     { 00440, AID_ROOT,      AID_SHELL,     0, "system/etc/init.trout.rc" },
diff --git a/rootdir/init.rc b/rootdir/init.rc
index e913dab..0205033 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -31,9 +31,6 @@ sysclktz 0
 
 loglevel 3
 
-# Redirect rilpoxy to rild
-    symlink /dev/socket/rild /dev/socket/rilproxy
-
 # Backward compatibility
     symlink /system/etc /etc
     symlink /sys/kernel/debug /d
@@ -573,3 +570,9 @@ service gonksched /system/bin/gonksched
 service b2g /system/bin/b2g.sh
     class main
     onrestart restart media
+
+service rilproxy /system/bin/rilproxy
+    class main
+    socket rilproxy stream 660 root system
+    user root
+    group radio
-- 
1.7.9.5

